features/networking/ovn-egress-ip.feature

```diff
@@ -482,3 +482,151 @@ Feature: OVN Egress IP related features
+    And I run the :create admin command with:
+      | f | egressip1.yaml |
+    Then the step should succeed
+
+    # Check source ip is egress ip
+    Then I wait up to 20 seconds for the steps to pass:
+    """
+    When I execute on the pod:
+      | curl | -s | --connect-timeout | 5 | <%= cb.ipecho_url %> |
+    Then the step should succeed
+    And the output should contain "<%= cb.valid_ip %>"
+    """
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-33710
+  @admin
+  @destructive
+  Scenario: An EgressIP object can not have multiple egress IP assignments on the same node
+    Given I store the schedulable workers in the :nodes clipboard
+    Then label "k8s.ovn.org/egress-assignable=true" is added to the "<%= cb.nodes[0].name %>" node
+
+    #Get unused IP as egress ip
+    Given I store a random unused IP address from the reserved range to the clipboard
+
+    #Create first egress ip object with two EgressIPs
+    When I obtain test data file "networking/ovn-egressip/egressip2.yaml"
+    And I replace lines in "egressip2.yaml":
+      | 172.31.249.227 | "<%= cb.valid_ips[0] %>" |
+      | 172.31.249.228 | "<%= cb.valid_ips[1] %>" |
+    And I run the :create admin command with:
+      | f | egressip2.yaml |
+    And admin ensures "egressip2" egress_ip is deleted after scenario
+
+    # Check only one IP assigned.
+    When I run the :get admin command with:
+      | resource       | egressip                    |
+      | resource_name  | egressip2                   |
+      | o              | jsonpath={.status.items[*]} |
+    Then the step should succeed
+    And evaluation of `@result[:response].chomp.match(/\d{1,3}\.\d{1,3}.\d{1,3}.\d{1,3}/).length` is stored in the :egressip_num clipboard
+    Then the expression should be true> cb.egressip_num == 1
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-33617
+  @admin
+  Scenario: Common user cannot tag the nodes by labelling them as egressIP nodes
+    Given I select a random node's host
+    And evaluation of `node.name` is stored in the :egress_node clipboard
+
+    #Label nodes with normal user
+    When I run the :label client command with:
+      | resource | node                               |
+      | name     | <%= cb.egress_node %>              |
+      | key_val  | k8s.ovn.org/egress-assignable=true |
+    Then the step should fail
+    And the output should match:
+      | nodes "<%= cb.egress_node %>" is forbidden |
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-33719
+  @admin
+  @destructive
+  Scenario: Any egress IP can only be assigned to one node only
+    Given I store the schedulable workers in the :nodes clipboard
+    Then label "k8s.ovn.org/egress-assignable=true" is added to the "<%= cb.nodes[0].name %>" node
+    And label "k8s.ovn.org/egress-assignable=true" is added to the "<%= cb.nodes[1].name %>" node
+
+    #Get unused IP as egress ip
+    Given I store a random unused IP address from the reserved range to the clipboard
```

Comment:
Does this step choose 1 IP or multiple IPs from the pool? Because I see `cb.valid_ips` is a plural and a list. If that's the case, then we should say `I store a few random unused IP addresses` (plural) right? Please correct me if I am wrong.

Reply:
@kedark3 you can look at the step_definiton to answer that question https://github.com/openshift/verification-tests/blob/master/features/step_definitions/networking.rb#L526

Reply:
This makes sense now. I think the step definition is poorly worded as it is actually returning a list (with 1 or more items) but the statement makes you think its just going to give you 1 address. But not the issue with this PR. Please resolve this discussion.

Reply:
@kedark3 did you see this line?  https://github.com/openshift/verification-tests/blob/master/features/step_definitions/networking.rb#L550. The step does just return just one random address.

Reply:
@pruan-rht I am confused due to this line : https://github.com/openshift/verification-tests/pull/1859/files#diff-cfc58074315d4bf1dad9d1c0d865a83552b43a6fd8740713758efc1d53204bd5R560
it is referencing ip address at the index [1] makes me think more than one IPs are returned by previous steps. But maybe `cb.valid_ips` is node ips then? sorry for the back and forth.

Reply:
@kedark3 `cb` is a global variable... `cb.valid_ips` is assigned in this line https://github.com/openshift/verification-tests/blob/master/features/step_definitions/networking.rb#L549 and the next line is ```cb[cb_name]=cb.valid_ips[0]```.  `cb_name` is defined [here](https://github.com/openshift/verification-tests/blob/master/features/step_definitions/networking.rb#L528)